docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,310 @@
+
+export default function HomeScreen() {
+  return (
+    <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
+      <TextInput placeholder="Type here..." />
+    </KeyboardAvoidingView>;
+  );
+}
+```
+
+In the above example, the height of the `KeyboardAvoidingView` automatically adjusts based on the device's keyboard height, which ensures that the input is always visible.
+
+When using a Bottom Tab navigator on Android, you might notice that focusing on an input field causes the bottom tabs to be pushed above the keyboard. To address this issue, add the `softwareKeyboardLayoutMode` property to your Android configuration in [app confg](/workflow/configuration/) and set it to `pan`.
+
+```json app.json
+"expo" {
+  "android": {
+    "softwareKeyboardLayoutMode": "pan"
+  }
+}
+```
+
+After adding this property, restart the development server and reload your app to apply the changes.
+
+It's also possible to hide the bottom tab when the keyboard opens using [`tabBarHideOnKeyboard`](https://reactnavigation.org/docs/bottom-tab-navigator/#tabbarhideonkeyboard). It is an option with the Bottom Tab Navigator. If set to `true`, it will hide the bar when the keyboard opens.
+
+```tsx app/_layout.tsx
+import { Tabs } from 'expo-router';
+
+export default function TabLayout() {
+  return (
+    <Tabs
+      screenOptions={{
+        tabBarHideOnKeyboard: true,
+      }}>
+      <Tabs.Screen name="index" />
+    </Tabs>
+  );
+}
+```
+
+### Keyboard events
+
+The `Keyboard` module from React Native allows you to listen for native events, react to them, and make changes to the keyboard, such as dismissing it.
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method accepts an event name and a callback function as arguments. When the keyboard is shown or hidden, the callback function is called with the event data.
+
+The following example illustrates a use case for adding a keyboard listener. The state variable `isKeyboardVisible` is toggled each time the keyboard shows or hides. Based on this variable, a button allows the user to dismiss the keyboard only if the keyboard is active. Also, notice that the button uses the `Keyboard.dismiss` method.
+
+```tsx HomeScreen.tsx
+import { useEffect, useState } from 'react';
+import { Keyboard, View, Button, TextInput } from 'react-native';
+
+export default function HomeScreen() {
+  const [isKeyboardVisible, setIsKeyboardVisible] = useState(false);
+
+  useEffect(() => {
+    const showSubscription = Keyboard.addListener('keyboardDidShow', handleKeyboardShow);
+    const hideSubscription = Keyboard.addListener('keyboardDidHide', handleKeyboardHide);
+
+    return () => {
+      showSubscription.remove();
+    };
+  }, []);
+
+  const handleKeyboardShow = event => {
+    setIsKeyboardVisible(true);
+  };
+
+  const handleKeyboardHide = event => {
+    setIsKeyboardVisible(false);
+  };
```

Comment:
Correct me if I'm wrong, but shouldn't these two either be defined inside the useEffect function or else added to the deps array?